### DISCUSSION TITLE
allow skip build by git commit message

### DISCUSSION THREAD

**Discussion by bluelovers** [vercel/community#60] [Help] [answered]
some ci support `[skip ci]` for skip ci task

i hope `now` also can have one, something like `[skip now]` or `[skip deploy]`

**Comment by coetry** [comment]
@bluelovers I'm not sure I exactly understand the reasoning behind this :smile: 

What action would you like to skip and why?

**Reply by bluelovers** [reply]
sometime , we just push new commit , but nothing is need change and no need re-build
this also can avoid waste now server spend time do it

**Reply by coetry** [reply]
@bluelovers we automatically de-dupe commits and if the contents haven't changed between commits, then the file hashes will remain the same and we won't waste any unneeded server resources to build :smile:

**Comment by remorses** [comment]
I have a Github action that pushes a commit to bump versions, this commit does not add any meaningful change, i would like to have a way to not trigger a redeploy

**Reply by styfle** [reply]
Some users would want a version bump to deploy in the case where the version in displayed in the footer of their website.

You could add your unused file to `.vercelignore` which would prevent deploying when that file changes.

https://vercel.com/guides/prevent-uploading-sourcepaths-with-vercelignore

**Reply by remorses** [reply]
Thank you for the reply

**Comment by stefanbuck** [comment]
Has this behaviour changed? Looks like files and folder listed in `.vercelignore` are still trigger a deploy. 

Example: A workflow runs whenever the vercel app triggers a `deployment_status` event with `state === 'success'`, meaning the last deploy is completed. This workflow takes a screenshot and commits this to the repo in a folder named `screenshots`. This older is listed in the `.vercelignore` file. However this screenshot commit generates a new deploy. This results in an endless, push deploy loop as you can see here https://github.com/stefanbuck/tentacle.app/actions?query=workflow%3AScreenshot

**Reply by stefanbuck** [reply]
I had another play with the vercel-cli and there the vercelignore file works as expected. Adding a new screenshot is not creating a new build. So maybe it's a "feature" within the Vercal GitHub App.

**Comment by styfle** [comment, marked as answer]
The [Ignore Build Step](https://vercel.com/docs/platform/projects#ignored-build-step) setting allows you to write arbitrary commands to skip the build.

In this case you could skip based on commit message.

See this [How do I use the "Ignored Build Step" field on Vercel?](https://vercel.com/knowledge/how-do-i-use-the-ignored-build-step-field-on-vercel) guide for an example.

**Reply by jamsinclair** [reply]
I found this didn't fully give an actionable example. (No discredit to the docs, just was hoping for an immediate solution).

For those looking for a `[ci skip]` or `[skip deploy]` like solution. I'm currently using the following command for Vercel's [Ignore Build Step](https://vercel.com/docs/platform/projects#ignored-build-step):
```
git log -1 --pretty=oneline --abbrev-commit | grep "[skip deploy]" || exit 1
```
_Note: adjust the `grep "[skip deploy]"` portion according to your needs._

**Reply by amyegan** [reply]
Thanks for sharing this solution @jamsinclair! We also have a guide on [using GitHub Actions with Vercel](https://vercel.com/support/articles/how-can-i-use-github-actions-with-vercel). Going that route could provide more control over Builds and Deployments from GitHub repos.

**Reply by b3nk3** [reply]
i've added this but my subsequent commits, on the same branch, without `[skip deploy]` are also cancelled...

![image](https://user-images.githubusercontent.com/10125940/190131455-356cefe7-4cbc-4cea-8bf5-e317fe51e744.png)

**Reply by b3nk3** [reply]
I believe it was because the left half of the ternary always returned something...

I rewrote it slightly and it made it work: `git log -1 --pretty=oneline --abbrev-commit | grep "[skip deploy]" && exit 1 || exit 0`

**Reply by stephenhmarsh** [reply]
I found that `grep "[skip deploy]"` gives me false positives on partial string matches.

It should call `grep -w` for word matching with a regex, no?

Also,[ according to the docs](https://vercel.com/guides/how-do-i-use-the-ignored-build-step-field-on-vercel), it should return `0` to **_skip_** the build and `1` to **_not_** skip?

So I ended up with:

```
git log -1 --pretty=oneline --abbrev-commit | grep -w "\[skip deploy\]" && exit 0 || exit 1
```
